httpx/dispatch/connection_pool.py

```diff
@@ -113,9 +113,13 @@ def num_connections(self) -> int:
         request: AsyncRequest,
         verify: VerifyTypes = None,
         cert: CertTypes = None,
-        timeout: TimeoutTypes = None,
+        timeout: TimeoutTypes = UNSET,
     ) -> AsyncResponse:
-        connection = await self.acquire_connection(origin=request.url.origin)
+        timeout = timeout if timeout is not UNSET else self.timeout
+
+        connection = await self.acquire_connection(
+            origin=request.url.origin, timeout=timeout
```

Comment:
Slight change from the previous code, now we use the provided `timeout` for acquiring the connection too.